crates/merod/src/cli/config.rs

```diff
@@ -1,95 +1,213 @@
+        /* 3. Pure printing (no edits)                                        */
+        /* ------------------------------------------------------------------ */
+        if !keys_to_print.is_empty() {
+            // Print only the requested sub-sections
+            let view = config.view_keys(&keys_to_print)?;
+            match self.print {
+                OutputFormat::Default | OutputFormat::Toml => {
+                    print_config(&view, PrintFormat::Toml)?
+                }
+                OutputFormat::Json => print_config(&view, PrintFormat::Json)?,
+            }
+        } else {
+            // Print the whole config
+            let full = config.as_map();
+            match self.print {
+                OutputFormat::Default | OutputFormat::Toml => {
+                    print_config(&full, PrintFormat::Toml)?
+                }
+                OutputFormat::Json => print_config(&full, PrintFormat::Json)?,
+            }
+        }
 
         Ok(())
     }
+}
 
-    pub fn validate_toml(self, doc: &toml_edit::DocumentMut) -> EyreResult<()> {
-        let tmp_dir = temp_dir();
-        let tmp_path = tmp_dir.join(CONFIG_FILE);
-
-        write(&tmp_path, doc.to_string())?;
-
-        let tmp_path_utf8 = Utf8PathBuf::try_from(tmp_dir)?;
-
-        drop(ConfigFile::load(&tmp_path_utf8)?);
+/// Wrapper for displaying config diffs in human-readable form
+pub struct Diff<'a>(pub &'a BTreeMap<String, (Option<String>, Option<String>)>);
 
+impl<'a> std::fmt::Display for Diff<'a> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        if self.0.is_empty() {
+            writeln!(f, "No changes detected.")?;
+            return Ok(());
+        }
+        for (key, (old_val, new_val)) in self.0 {
+            match (old_val, new_val) {
+                (Some(old), Some(new)) if old != new => {
+                    writeln!(
+                        f,
+                        "{} {} {}",
+                        key.green().bold(),
+                        "changed from".yellow(),
+                        old.red()
+                    )?;
+                    writeln!(f, "{} {}", "to".yellow(), new.green())?;
+                }
+                (None, Some(new)) => {
+                    writeln!(
+                        f,
+                        "{} {} {}",
+                        key.green().bold(),
+                        "set to".yellow(),
+                        new.green()
+                    )?;
+                }
+                (Some(old), None) => {
+                    writeln!(
+                        f,
+                        "{} {} {}",
+                        key.green().bold(),
+                        "removed (was)".yellow(),
+                        old.red()
+                    )?;
+                }
+                _ => {
+                    // No change or unknown case - skip printing
+                }
+            }
+        }
```

Comment:
can you show what this looks like?

"Consider printing diff(1)-style diffs, which may be facilitated by similar"

https://github.com/calimero-network/core/issues/1069#:~:text=Consider%20printing%20diff(1)%2Dstyle%20diffs%2C%20which%20may%20be%20facilitated%20by%20similar